pkg/daemon/update.go

```diff
@@ -82,25 +82,28 @@ func (dn *Daemon) updateOSAndReboot(newConfig *mcfgv1.MachineConfig) error {
 
 		dn.recorder.Eventf(node, corev1.EventTypeNormal, "Drain", "Draining node to update config.")
 
-		err = wait.ExponentialBackoff(wait.Backoff{
+		backoff := wait.Backoff{
 			Steps:    5,
 			Duration: 10 * time.Second,
 			Factor:   2,
-		}, func() (bool, error) {
+		}
+		var lastErr error
+		wait.ExponentialBackoff(backoff, func() (bool, error) {
 			err := drain.Drain(dn.kubeClient, []*corev1.Node{node}, &drain.DrainOptions{
 				DeleteLocalData:    true,
 				Force:              true,
 				GracePeriodSeconds: 600,
 				IgnoreDaemonsets:   true,
 			})
 			if err != nil {
-				glog.Infof("Draining failed with: %v; retrying...", err)
+				lastErr = err
 				return false, nil
 			}
+			lastErr = nil
 			return true, nil
 		})
-		if err != nil {
-			return err
+		if lastErr != nil {
+			return errors.Wrapf(lastErr, "Failed to drain node (%s tries)", backoff.Steps)
 		}
```

Comment:
we're gonna miss the `wait.*` error this way cause other err may happen other than just timeout
```
// If the condition never returns true, ErrWaitTimeout is returned. All other
// errors terminate immediately.
func ExponentialBackoff(backoff Backoff, condition ConditionFunc) error {
```
I was thinking something like https://github.com/openshift/machine-config-operator/blob/master/pkg/operator/sync.go#L357-L362

Reply:
this is not blocking of course, we control the ConditionFunc here